packages/web/src/components/test-table/TestTable.tsx

```diff
@@ -0,0 +1,289 @@
+  wrapperClassName,
+  tableClassName,
+  getRowClassName,
+  defaultSorter,
+  onClickRow,
+  onSort,
+  onReorder,
+  isPaginated = false,
+  isReorderable = false,
+  isVirtualized = false,
+  maxRowNum = 20
+}: TestTableProps) => {
+  const defaultColumn = useMemo(
+    () => ({
+      // Default resizing column props
+      minWidth: 64,
+      width: 64,
+      maxWidth: 200
+    }),
+    []
+  )
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    state: { sortBy }
+  } = useTable(
+    {
+      columns,
+      data,
+      defaultColumn,
+      autoResetSortBy: false,
+      autoResetResize: false,
+      manualSortBy: true
+    },
+    useSortBy,
+    useResizeColumns,
+    useFlexLayout
+  )
+
+  // NOTE: react-table allows for multple sorters, but we are only checking the first here
+  // - This can be updated if we need multiple sorters in the future
+  const handleSortChange = useCallback(() => {
+    let sorter = null
+    let order = 'ascend'
+
+    if (sortBy.length === 0) {
+      // Use defaultSorter if sortBy array is empty
+      sorter = defaultSorter
+    } else {
+      // Use the sorter from the column
+      const sortColumn = columns.find((c) => c.id === sortBy[0].id)
+      sorter = sortColumn?.sorter
+      order = sortBy[0]?.desc ? 'descend' : 'ascend'
+    }
+
+    if (sorter) {
+      onSort?.({ column: { sorter }, order })
+    }
+  }, [columns, defaultSorter, onSort, sortBy])
+
+  useEffect(handleSortChange, [handleSortChange, sortBy])
+
+  const renderHeader = useCallback(() => {
+    return headerGroups.map((headerGroup) => (
+      <tr
+        className={styles.tableHeadRow}
+        {...headerGroup.getHeaderGroupProps()}
+        key={headerGroup.id}
+      >
+        {headerGroup.headers.map((column) => (
+          <th
+            className={cn(styles.tableHeader, {
+              [styles.titleHeader]: Boolean(column.accessor),
+              [styles.hasSorter]: column.disableSortBy !== true,
+              [styles.leftAlign]: column.align === 'left',
+              [styles.rightAlign]: column.align === 'right'
+            })}
+            {...column.getHeaderProps()}
+            key={column.id}
+          >
+            {/* Sorting Container */}
+            <div {...column.getSortByToggleProps()}>
+              <div className={styles.textCell}>{column.render('Header')}</div>
+              {!column.disableSortBy ? (
+                <div className={styles.sortCaretContainer}>
+                  {!column.isSorted || !column.isSortedDesc ? (
+                    <IconCaretUp className={styles.sortCaret} />
+                  ) : null}
+                  {!column.isSorted || column.isSortedDesc ? (
+                    <IconCaretDown className={styles.sortCaret} />
+                  ) : null}
+                </div>
+              ) : null}
+            </div>
+            {/* Resizing Container */}
+            {!column.disableResizing && (
```

Comment:
I think we decided on using ternaries with the false case being `null` instead of `&&` in jsx